src/VecSim/algorithms/hnsw/hnsw_batch_iterator.h

```diff
@@ -12,34 +12,38 @@ using spaces::dist_func_t;
 
 template <typename DataType, typename DistType>
 class HNSW_BatchIterator : public VecSimBatchIterator {
-private:
+protected:
     HNSWIndex<DataType, DistType> *index;
     dist_func_t<DistType> dist_func;
     size_t dim;
     VisitedNodesHandler *visited_list; // Pointer to the hnsw visitedList structure.
     tag_t visited_tag;                 // Used to mark nodes that were scanned.
     idType entry_point;                // Internal id of the node to begin the scan from.
     bool depleted;
-    size_t orig_ef_runtime; // Original default parameter to reproduce.
+    size_t ef; // EF Runtime value for this query.
 
     // Data structure that holds the search state between iterations.
-    using candidatesMinHeap = vecsim_stl::min_priority_queue<DistType, idType>;
-    using candidatesMaxHeap = vecsim_stl::max_priority_queue<DistType, idType>;
+    template <typename Value>
```

Comment:
i find value a bit confusing as a name to the identifier (I would expect that the distance will be called "the value")